### PR TITLE
[Community] Add 'Projects/' folder, and the first example project.

### PR DESCRIPTION
**Pull request by RangiLyu**
# Motivation
Introducing new models and features into OpenMMLab's algorithm libraries has long been complained to be troublesome due to the rigorous requirements on code quality, which could hinder the fast iteration of SOTA models and might discourage potential contributors from sharing their latest outcome here.

# Modification
This PR adds a new projects/ folder, which will be a place for some experimental models/features. Implementations inside might be not quite perfect but already fine to produce some exciting results. We hope that this PR can help us better embrace the contribution from our community. We also add the first example project to illustrate what we expect a good project to have.

### CODE REVIEW

File: projects/example_project/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  - [ ] Training-time correctness
+
+    <!-- If you are reproducing the result from a paper, checking this item means that you should have trained your model from scratch based on the original paper's specification and verified that the final result matches the report within a minor error range. -->
+
+- [ ] Milestone 3: Good to be a part of our core package!
+
+  - [ ] Type hints and docstrings
+
+    <!-- Ideally *all* the methods should have [type hints](https://www.pythontutorial.net/python-basics/python-type-hints/) and [docstrings](https://google.github.io/styleguide/pyguide.html#381-docstrings). [Example](https://github.com/open-mmlab/mmdetection/blob/5b0d5b40d5c6cfda906db7464ca22cbd4396728a/mmdet/datasets/transforms/transforms.py#L41-L169) -->
```
Comment (hhaAndroid) [inline projects/example_project/README.md:92]: https://github.com/open-mmlab/mmdetection/blob/5b0d5b40d5c6cfda906db7464ca22cbd4396728a/mmdet/datasets/transforms/transforms.py   

This link can be replaced

File: projects/example_project/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  - [ ] Unit tests
+
+    <!-- Unit tests for each module are required. [Example](https://github.com/open-mmlab/mmdetection/blob/5b0d5b40d5c6cfda906db7464ca22cbd4396728a/tests/test_datasets/test_transforms/test_transforms.py#L35-L88) -->
```
Comment (hhaAndroid) [inline projects/example_project/README.md:96]: same

File: projects/example_project/README.md (hunk 1 above)
Comment (RangiLyu) [reply projects/example_project/README.md:92]: Replace with what?

Comment (hhaAndroid) [reply projects/example_project/README.md:92]: https://github.com/open-mmlab/mmdetection/blob/3.x/mmdet/datasets/transforms/transforms.py

Comment (RangiLyu) [reply projects/example_project/README.md:92]: > https://github.com/open-mmlab/mmdetection/blob/3.x/mmdet/datasets/transforms/transforms.py

If the code is updated, the line permalink will mismatch. It must link to a specific commit hash.

File: projects/example_project/README.md (hunk 2 above)
Comment (ZwwWayne) [reply projects/example_project/README.md:96]: I think we could use permlink

File: projects/example_project/README.md (hunk 1 above)
Comment (ZwwWayne) [reply projects/example_project/README.md:92]: I think we can keep the current link for now.

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**